packages/SwingSet/src/netstring.js

```diff
@@ -0,0 +1,81 @@
+// adapted from 'netstring-stream', https://github.com/tlivings/netstring-stream/
+import { assert } from '@agoric/assert';
+
+const { Transform } = require('stream');
+
+const COLON = 58;
+const COMMA = 44;
+
+// input is a Buffer, output is a netstring-wrapped Buffer
+export function encode(data) {
+  const prefix = Buffer.from(`${data.length}:`);
+  const suffix = Buffer.from(',');
+  return Buffer.concat([prefix, data, suffix]);
+}
+
+// input is a sequence of strings, output is a byte pipe
+export function createWriter() {
+  function transform(chunk, encoding, callback) {
+    assert.equal(encoding, 'buffer');
+    callback(encode(chunk));
+  }
+  return new Transform({ transform });
+}
+
+// Input is a Buffer containing zero or more netstrings and maybe some
+// leftover bytes. Output is zero or more decoded Buffers, one per netstring,
```

Comment:
Might be clearer if s/decoded buffers/strings.

Reply:
but I don't think they are `string`s, right? they're strictly `Buffer` objects, if I understand correctly

Reply:
Ah, you are right. I was confused by the symmetry and “decoded”. Would “extracted” be a better term? Decode implied bytes in, string out to me.